RNTester/RNTester-macOS/TurboModuleCxx/TurboModulesProvider.cpp

```diff
@@ -0,0 +1,304 @@
+                  VerifyElseCrash(count >= 2);
+                  VerifyElseCrash(args[count - 1].isObject() && args[count - 1].asObject(runtime).isFunction(runtime));
+                  VerifyElseCrash(args[count - 2].isObject() && args[count - 2].asObject(runtime).isFunction(runtime));
+                  serializableArgumentCount -= 2;
+                  break;
+                case MethodReturnType::Void:
+                case MethodReturnType::Promise:
+                  // handled below
+                  break;
+              }
+              auto argReader = winrt::make<JsiReader>(runtime, args, serializableArgumentCount);
+
+              // prepare output value
+              // TODO: it is no reason to pass a argWriter just to receive [undefined] for void, should be optimized
+              auto argWriter = winrt::make<JsiWriter>(runtime);
+
+              // call the function
+              switch (method.ReturnType) {
+                case MethodReturnType::Void: {
+                  method.Method(argReader, argWriter, nullptr, nullptr);
+                  return facebook::jsi::Value::undefined();
+                }
+                case MethodReturnType::Promise: {
+                  return facebook::react::createPromiseAsJSIValue(
+                      runtime, [=](facebook::jsi::Runtime &runtime, std::shared_ptr<facebook::react::Promise> promise) {
+                        method.Method(
+                            argReader,
+                            argWriter,
+                            [promise, &runtime](const IJSValueWriter &writer) {
+                              auto result = writer.as<JsiWriter>()->MoveResult();
+                              promise->resolve(result);
+                            },
+                            [promise, &runtime](const IJSValueWriter &writer) {
+                              auto result = writer.as<JsiWriter>()->MoveResult();
+                              VerifyElseCrash(result.isString());
+                              promise->reject(result.getString(runtime).utf8(runtime));
+                            });
+                      });
+                }
+                case MethodReturnType::Callback:
+                case MethodReturnType::TwoCallbacks: {
+                  facebook::jsi::Value resolveFunction;
+                  facebook::jsi::Value rejectFunction;
+                  if (method.ReturnType == MethodReturnType::Callback) {
+                    resolveFunction = {runtime, args[count - 1]};
+                  } else {
+                    resolveFunction = {runtime, args[count - 2]};
+                    rejectFunction = {runtime, args[count - 1]};
+                  }
+
+                  auto makeCallback = [&runtime](
+                                          const facebook::jsi::Value &callbackValue) noexcept->MethodResultCallback {
+                    // workaround: xcode doesn't accept a captured value with only rvalue copy constructor
```

Comment:
This is a 10.13 deployment target workaround, right? If so, would be good to reword this to clarify the actual problem.